### PR TITLE
Adding loader for acoustic brainz genre dataset

### PR DESCRIPTION
**Pull request by PRamoneda**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the size of the datasets (e.g. number files and duration, hours)
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of music included in the dataset
- [x] Indicate any relevant papers related to the dataset

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `mirdata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in mirdata, e.g. `mirdata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/mirdata.rst` and `docs/source/datasets.rst`
- [x] Add the module to `mirdata/__init__.py`
- [x] Add the module to the list in the `README.md` file, section `Currently supported datasets`
- [x] Docs building
- [x] Run and pass `test_full_dataset` for Acousticbrainzgenre dataset

#### Please-do-not-edit flag
To reduce friction, we will make commits on top of contributor's pull requests by default unless they use the `please-do-not-edit` flag. If you don't want this to happen don't forget to add the flag when you start your pull request.

### CODE REVIEW

File: mirdata/acousticbrainz_genre.py
Comment (alastair) [inline mirdata/acousticbrainz_genre.py:69]: is this a musicbrainz recording id? If so, it would be good to be explicit about this, because in Musicbrainz, a "track" is a different thing

File: mirdata/acousticbrainz_genre.py
Comment (alastair) [inline mirdata/acousticbrainz_genre.py:66]: Please be careful about the naming of this, it's _not_ a general AcousticBrainz dataset.

File: mirdata/acousticbrainz_genre.py
Comment (alastair) [inline mirdata/acousticbrainz_genre.py:113]: the only required data in acousticbrainz is the recording id. These items should use `.get()` to handle missing items.

Comment (alastair) [reply mirdata/acousticbrainz_genre.py:113]: These cached properties load the file each time that you call a different one (the `load_extractor`). Maybe it would make more sense to cache the result of that method instead

Comment (alastair) [reply mirdata/acousticbrainz_genre.py:113]: in this dataset, the value of `metadata.recording_id` is not always the same as the recording id used in the filename (what you call track_id here I think). Because of this, I think that there should also be a metadata property to load this metadata field.

File: mirdata/acousticbrainz_genre.py
Comment (alastair) [inline mirdata/acousticbrainz_genre.py:3624]: Genre?

File: mirdata/acousticbrainz_genre.py
Comment (alastair) [inline mirdata/acousticbrainz_genre.py:3612]: not correct

File: mirdata/acousticbrainz_genre.py
Comment (alastair) [inline mirdata/acousticbrainz_genre.py:3648]: This dataset has no audio, so I think it would make sense to not add this item in the documentation

File: mirdata/acousticbrainz_genre.py
Comment (alastair) [inline mirdata/acousticbrainz_genre.py:3795]: remove

File: scripts/make_acousticbrainz_genre.py
Comment (alastair) [inline scripts/make_acousticbrainz_genre.py:51]: Is there a reason to write json by hand here? I think it would be safer to write it as json, unless there's a specific reason.

File: mirdata/acousticbrainz_genre.py
Comment (PRamoneda) [reply mirdata/acousticbrainz_genre.py:3648]: it is copy paste from other dataset.. I havent developed the download method yet

File: mirdata/acousticbrainz_genre.py
Comment (PRamoneda) [reply mirdata/acousticbrainz_genre.py:3612]: it is copy paste too...

File: scripts/make_acousticbrainz_genre.py
Comment (PRamoneda) [reply scripts/make_acousticbrainz_genre.py:51]: Yes. The full index is massive, and python throws an exception due to that. The rest of datasets have been developed with `JSON` package

File: mirdata/acousticbrainz_genre.py
Comment (PRamoneda) [reply mirdata/acousticbrainz_genre.py:69]: `track id` are the keys in the index (of 5 million elements). Each key is composed of all the information of each track that was in the .tsv files on Zenodo for each "mini - dataset" (allmusic, tagtraum, discogs and lastfm)

File: mirdata/acousticbrainz_genre.py
Comment (PRamoneda) [reply mirdata/acousticbrainz_genre.py:66]: nit done

File: mirdata/acousticbrainz_genre.py
Comment (PRamoneda) [reply mirdata/acousticbrainz_genre.py:3624]: nit - In this dataset this message is no needed

File: mirdata/acousticbrainz_genre.py
Comment (PRamoneda) [reply mirdata/acousticbrainz_genre.py:3612]: nit done

File: tests/test_utils.py
Comment (nkundiushuti) [inline tests/test_utils.py:88]: I think this was a part of another PR which is currently pending approval. so you should remove this file before merging to master

Comment (PRamoneda) [reply tests/test_utils.py:88]: I think that the other PR have to be merged first and then with this PR we won't have any problem :)

File: docs/source/datasets.rst
Comment (nkundiushuti) [inline docs/source/datasets.rst:28]: is this the name of the module in mirdata or acousticbrainz_genre? was it shortened because the table would overflow?

**Review by nkundiushuti — APPROVED** [review]

File: tests/test_utils.py
Comment (magdalenafuentes) [reply tests/test_utils.py:88]: I think @nkundiushuti might be right, in any case check if this is a conflict and if it is just remove this, and merge master to have the change integrated

File: docs/source/datasets.rst
Comment (magdalenafuentes) [reply docs/source/datasets.rst:28]: Mmmm same concern, it would be good to keep consistency naming the loader. Is it much longer than others? e.g. 
`medleydb_solos_db`
`acousticbrainz_genre`
`acousticbrainz_gen`

**Comment by magdalenafuentes** [comment]
@PRamoneda looks good to me, see a few comments above. Then you can update to master after we merge #352 and go ahead and merge, does that sound good?

**Comment by magdalenafuentes** [comment]
And there's a small conflict with `datasets.rst`

File: docs/source/datasets.rst
Comment (PRamoneda) [reply docs/source/datasets.rst:28]: yes it is bigger.... but well I am going to keep consistency... and extend the table spaces... thanks!

**Comment by magdalenafuentes** [comment]
@PRamoneda looking at the table I noticed that in the `acousticbrainz_genre` column the `audio` field was switched by `features`. I'm more inclined to keep `audio` and add `feature` as an `emoji` instead (as 📺, 🔑), what do you think?

**Comment by magdalenafuentes** [comment]
Mmm I'm seeing same error as in #352, I guess both are from an outdated version of master? Should be fixed in #352 and then merged here

**Pull request merged**
